digdag-standards/src/main/java/io/digdag/standards/command/kubernetes/KubernetesClientConfig.java

```diff
@@ -19,42 +19,50 @@ public static KubernetesClientConfig create(final Optional<String> name,
             final Config systemConfig,
             final Config requestConfig)
     {
-        if (requestConfig.has("kubernetes")) {
-            // from task request config
-            return KubernetesClientConfig.createFromTaskRequestConfig(name, requestConfig.getNested("kubernetes"));
-        }
-        else {
-            // from system config
-            return KubernetesClientConfig.createFromSystemConfig(name, systemConfig);
+
+        String clusterName = null;
+        if (name.isPresent()) {
+            clusterName = name.get();
         }
-    }
 
-    private static KubernetesClientConfig createFromTaskRequestConfig(final Optional<String> name,
-            final Config config)
-    {
-        // TODO
-        // We'd better to customize cluster config by task request config??
-        throw new UnsupportedOperationException("Not support yet");
-    }
+        Config extractedSystemConfig = null;
+        if (systemConfig != null && systemConfig.get("agent.command_executor.type", String.class, "").equals("kubernetes")) {
```

Comment:
This code assume we switch CommandExecutor by "agent.command_executor.type" but we don't implement switching mechanism for CommandExecutor.